internal/operator/subscription.go

```diff
@@ -36,36 +36,26 @@ func (c operatorClient) GetSubscriptionData(options OperatorCheckOptions) ([]Sub
 	SubscriptionList := []SubscriptionData{}
 
 	for _, pkgm := range packageManifests.Items {
-		// iterate through the channels and set value
-		// for defaultChannel variable
-		var defaultChannel pkgserverv1.PackageChannel
-		for _, pkgch := range pkgm.Status.Channels {
-			if pkgch.IsDefaultChannel(pkgm) {
-				defaultChannel = pkgch
-			}
-		}
-
-		// create a new subscription for each of the supported install mode types
-		// on the default channel
-		for _, installMode := range defaultChannel.CurrentCSVDesc.InstallModes {
-			if installMode.Supported {
-				SubscriptionList = append(SubscriptionList,
-					SubscriptionData{
-						Name:                   strings.Join([]string{defaultChannel.Name, pkgm.Name, strings.ToLower(string(installMode.Type)), "subscription"}, "-"),
-						Channel:                defaultChannel.Name,
-						CatalogSource:          options.CatalogSource,
-						CatalogSourceNamespace: options.CatalogSourceNamespace,
-						Package:                pkgm.Name,
-						InstallModeType:        installMode.Type,
-						InstallPlanApproval:    operatorv1alpha1.ApprovalAutomatic,
-					},
-				)
-			}
-
-			// Return subscriptions for all install modes
-			// only when AllInstallModes is true
-			if !options.AllInstallModes {
-				break
+		if pkgm.Status.CatalogSource == catalogSource {
```

Comment:
Not a blocker. This could have a lot of the indentation removed by switching up some of the logic.

```suggestion
		if pkgm.Status.CatalogSource != catalogSource {
		  continue
		}
		for _, pkgch := range pkgm.Status.Channels {
		  if !pkgch.IsDefaultChannel(pkgm) { continue }
```

etc...

Reply:
That's right. I gets a bit better to read. Here it is [349c93a](https://github.com/opdev/opcap/pull/188/commits/349c93a3fc416252b285064244cbe82d51ccbc2b)